src/codeu/chat/server/Server.java

```diff
@@ -193,6 +213,23 @@ public void run() {
         timeline.scheduleIn(RELAY_REFRESH_MS, this);
       }
     });
+    //Save the data periodically
+    this.timeline.scheduleNow(new Runnable() {
+      @Override
+      public void run() 
+      {
+        try
+        {
+          localFile.saveData();
+          timeline.scheduleIn(LOCAL_FILE_REFRESH_MS, this);
```

Comment:
If the above line throws an exception, this won't be rescheduled. This line should be executed inside a `finally` block to make sure that you always schedule the next refresh.